extension/background.js

```diff
@@ -1,112 +1,49 @@
-var windowCount = 'windowCount',
-	incogHistory = 'incogHistory',
-	incogRecent = 'incogRecent';
+if (!chrome.extension.inIncognitoContext) window.close();
 
-chrome.runtime.onMessage.addListener(function (request, sender, sendResponse) {
+var tabs = JSON.parse(localStorage.tabs || '{}'), incHist = JSON.parse(localStorage.incHist || '[]'), incRecent = JSON.parse(localStorage.incRecent || '[]');
 
-	if (!sender.tab.incognito) {
-		return;
-	}
+chrome.runtime.onSuspend.addListener(function(){
+	localStorage.setItem('tabs', JSON.stringify(tabs));
+	localStorage.setItem('incHist', JSON.stringify(incHist));
+	localStorage.setItem('incRecent', JSON.stringify(incRecent));
+	});
 
-	var tabInfo = {
-		id: sender.tab.id,
-		url: sender.tab.url,
-		title: sender.tab.title,
-		favIcon: sender.tab.favIconUrl,
-		timestamp: Date()
-	};
+chrome.tabs.onUpdated.addListener(function(tabId, chg, tab){
+	if ( ['chrome://newtab/', 'about:blank'].includes(tab.url) ) return;
 
-	if (localStorage.getItem(windowCount) == null) {
-		count = 1;
-		localStorage.setItem(windowCount, count);
-	}
-
-	if (localStorage.getItem(incogHistory) == null) {
-		var arr = [];
-		localStorage.setItem(incogHistory, JSON.stringify(arr));
-	}
-
-	if (localStorage.getItem(incogRecent) == null) {
-		var arr = [];
-		arr.length = 0;
-		localStorage.setItem(incogRecent, JSON.stringify(arr));
-	}
-
-	tab.storeData(tabInfo);
-
-});
-
-var tab = {
-	storeData: function (tabInfo) {
-		var arr = [];
-		arr = JSON.parse(localStorage.getItem(incogHistory));
-		arr.push(tabInfo);
-
-		localStorage.setItem(incogHistory, JSON.stringify(arr));
-	}
-}
-
-
-function recentRecord(tabId) {
-	var recent = [],
-		history = [],
-		historyLength;
-
-	history = JSON.parse(localStorage.getItem(incogHistory));
-	historyLength = history.length - 1;
-
-	if (tabId != undefined) {
-		var sameTab = [],
-			sameTabLength;
-
-		recent = JSON.parse(localStorage.getItem(incogRecent));
-
-		for (var i = 0; i <= historyLength; i++) {
-			if (tabId == history[i].id)
-				sameTab.push(history[i]);
+	var t, oldU, n;

```

Comment:
Rename these variables for better readability.

Reply:
I used `t` for "tab". Like `i` `j` `k` for loops. And `n` for "number".
Feel free to make it more readable.

Reply:
Sure.

Reply:
I was thinking of using meaningful names during development for better readability and minifying them before publishing it because right now un-minified code gets published on Chrome store.